variables.tf

```diff
@@ -53,6 +53,18 @@ variable "enabled_metrics" {
   ]
 }
 
+variable "create_iam_role" {
```

Comment:
Is there ever a situation where we don't want a role to be created, and we also don't want to specify an `iam_role_arn`? If not, maybe we should just have a single variable `iam_role_arn`, and create a local for `create_iam_role` that's true if no role ARN is specified?

I don't hugely mind either way, so if there's a reason you've done it like this that's fine. Just wanted to ask the question.

Reply:
I added `create_iam_role` to work around a chicken and egg issue that resulted in this error:

> Error: Invalid count argument
│ 
│   on ../../iam.tf line 2, in resource "aws_iam_role" "this":
│    2:   count = var.iam_role_arn != "" ? 0 : 1
│ 
│ The "count" value depends on resource attributes that cannot be determined
│ until apply, so Terraform cannot predict how many instances will be
│ created. To work around this, use the -target argument to first apply only
│ the resources that the count depends on.

I noticed that the AWS ASG module that we use [implemented this work around](https://registry.terraform.io/modules/terraform-aws-modules/autoscaling/aws/6.7.1#input_create_iam_instance_profile) and I implemented it in your module.

Reply:
Gotcha - that's annoying, but it makes perfect sense.